docker/studyu-cli.sh

```diff
@@ -0,0 +1,479 @@
+#!/bin/bash
+
+SCRIPT_DIR=$( cd -- "$( dirname -- "${BASH_SOURCE[0]}" )" &> /dev/null && pwd )
+PARENT_DIR=$(dirname "$SCRIPT_DIR")
+
+# Color codes for better output
+green='\e[0;32m'
+yellow='\e[1;33m'
+red='\e[0;31m'
+bold='\e[1m'
+reset='\e[0m'
+
+# Configuration file path
+CONFIG_FILE="$SCRIPT_DIR/.studyu_config"
+
+# Dictionary to map components to Docker Compose files
+declare -A compose_files=(
+  ["supabase-db"]="supabase/docker-compose-db.yml"
+  ["supabase"]="supabase/docker-compose.yml"
+  ["proxy"]="nginx/docker-compose-proxy.yml"
+  # ["studyu"]="studyu/docker-compose.yml"
+  ["studyu-app"]="studyu/docker-compose-app.yml"
+  ["studyu-designer"]="studyu/docker-compose-designer.yml"
+)
+available_components="${bold}Available components:\n${reset}"
+available_components+="  supabase-db        Postgres database used as Supabase backend\n"
+available_components+="  supabase           Supabase middleware (needs supabase-db)\n"
+available_components+="  proxy              Nginx reverse proxy to access the Supabase frontend if no StudyU component is started\n"
+# available_components+="  studyu             StudyU App and StudyU Designer (includes proxy)\n"
+available_components+="  studyu-app         StudyU App (includes proxy)\n"
+available_components+="  studyu-designer    StudyU Designer (includes proxy)"
+
+# Function to read configuration values from the file
+read_config() {
+  local failOnError="${1:-true}"
+
+  if [ "$failOnError" = true ] && [ ! -f "$CONFIG_FILE" ]; then
```

Comment:
the command `[` has an `-a` flag for logical `and`s

```suggestion
  if [ "$failOnError" = true -a ! -f "$CONFIG_FILE" ]; then
```

or, better, use bash's `[[`

```suggestion
  if [[ "$failOnError" = true && ! -f "$CONFIG_FILE" ]]; then
```

Same for all the other conditions; when we don't need POSIX compliance (e.g. writing a bash or zsh script) it's usually easier to work with `[[`

Reply:
I think I changed them all 👍